packages/entity-server/src/app/createApp.ts

```diff
@@ -22,8 +31,28 @@ import { attachRelationsContext } from '../routes/hierarchy/relations/middleware
 export function createApp() {
   return new MicroServiceApiBuilder(new TupaiaDatabase())
     .useBasicBearerAuth('entity-server')
-    .use<HierarchyRequest>('/v1/hierarchy/:hierarchyName/:entityCode', attachContext)
-    .get<HierarchyRequest>(
+    .use<SingleEntityRequest | MultiEntityRequest>(
+      '/v1/hierarchy/:hierarchyName',
+      attachCommonContext,
+    )
+    .use<MultiEntityRequest>(
+      ['/v1/hierarchy/:hierarchyName/relations', '/v1/hierarchy/:hierarchyName/descendants'],
```

Comment:
Have to be a bit explicit here as matching against `/v1/hierarchy/:hierarchyName` will match all routes which isn't what we want. We probably ought to switch to using regexes throughout this whole file to make the routes very explicit but I couldn't quite be bothered doing that at the moment

Reply:
No worries!

Reply:
The combination of generics + many-to-many mapping between `use` and `get` makes this section a bit harder to read now, but I'm not sure whether there is a better way...

I sense that it would be advantageous to see the supported routes with one glance,. Perhaps we will attempt to infer the request context in the future? For the time being, your approach seems the way to go.

One idea is grouping `use` and `get` statements together, although this may not actually improve things - still, probably worth a quick try.

Reply:
Could do something like this if you find it more readable?
```
.use<MultiEntityRequest>('/v1/hierarchy/:hierarchyName/relations', attachMultiEntityContext)
.use<MultiEntityRequest>('/v1/hierarchy/:hierarchyName/descendants', attachMultiEntityContext)
...
```

Alternatively, we could scrap the whole multi-entity/single-entity distinction and just make `entityCode` an optional param that overrides the `entities` param if it's present. That would achieve the same behaviour basically. We could even consolidate the routes then if we wish? I'm in two minds about such an approach; on the one hand it would remove complexity and reduce code duplication, on the other hand my gut does feel a bit that we want to treat the single entity and multi-entity cases as separate...

Thoughts on this?

Reply:
My gut also feels that the routes should be different, even if they share they same functionality under the hood... Also I think I find your suggestion above more readable. Finally, it's also a formatting thing... I wish that all those could fit in one line 😃

Reply:
I think I'll add a `setVersion` property to the api-builder so that it can prefix each endpoint with the version which should shave off a whole 4 chars per line 😝

Reply:
Haha, honestly, don't worry too much about it 😛

Reply:
I mean, if we can find ways to make this very easy to parse, it would be awesome, but not a biggie

Reply:
Also, could omit the `Entity` part of `MultiEntity...Request`/`MultiEntity...Route`, eg. `MultiDescendantsRequest`? Doesn't quite make as much sense to me tho...

Reply:
Better to leave `Entity` there, I reckon... Again, no stress about making this more succinct